hibernate-core/src/main/java/org/hibernate/type/descriptor/java/EnumJavaType.java

```diff
@@ -304,4 +305,21 @@ private <R> Set<R> valueSet(JdbcType jdbcType, BasicValueConverter<T,R> converte
 			}
 		}
 	}
+
+	@Override
+	public boolean equals(Object obj) {
+		if ( obj == this ) {
+			return true;
+		}
+		if ( obj == null || obj.getClass() != this.getClass() ) {
+			return false;
+		}
+		var that = (EnumJavaType<T>) obj;
+		return Objects.equals( this.getTypeName(), that.getTypeName() );
+	}
+
+	@Override
+	public int hashCode() {
+		return this.getTypeName().hashCode();
+	}
```

Comment:
This seems a bit strange. We don't usually have a well-defined notion of equality for `JavaType`s. What's the reason it's needed here?

Reply:
I noticed that in the `Map<JavaType<?>, BasicType<?>>` for the enums in the test new (`EnumJavaType`, `BasicType`) entries were being added for each run of the native query ( IIRC these were being created as `new EnumJavaType()` ).
Either way, it has always been my understanding that if you want to use something as key in a Map structure, it's almost always a good idea to implement equals() and hashCode() ?
I mean, I *may* have missed a more profound underlying cause, but I'm not sure that the affirmation that equality is usually not well defined actually holds true for enums?

Reply:
> I can affirm that there was a continued growth in the size of the BasicTypeRegistry.registryValues collection.

Ah so there *is* a place where we use `JavaType`s as keys!

I don't believe we should be doing that, since `equals()`/`hashCode()` is just not part of the contract for `JavaType`.

@sebersole WDYT?

Reply:
> the BasicTypeRegistry.registryValues collection

So I **completely deleted** that collection, and all the code which uses it, and _only one test failed_ (`ArrayPositionTest`).

So maybe that collection is not really all that important. I have not dig deeper to verify this.

Reply:
I can have a look ... on monday ;)
My first knee-jerk reaction when trying to fix a bug is also not to go in and delete the parts of the code that seem to be the culprit :grin: